components/eventing-controller/pkg/backend/eventmesh/eventmesh.go

```diff
@@ -269,6 +287,67 @@ func (em *EventMesh) handleCreateEventMeshSub(eventMeshSub *types.Subscription,
 	return eventMeshServerSub, nil
 }
 
+// handleWebhookAuthChange handles the EventMesh subscription WebhookAuth changes.
+// It uses the PATCH request API provided by EventMesh to update the subscription WebhookAuth.
+func (em *EventMesh) handleWebhookAuthChange(eventMeshSub *types.Subscription,
+	kymaSub *eventingv1alpha2.Subscription) error {
+	hash, err := backendutils.GetWebhookAuthHash(eventMeshSub.WebhookAuth)
+	if err != nil {
+		return fmt.Errorf("failed to get the EventMesh WebhookAuth hash: %w", err)
+	}
+
+	// skip if the WebhookAuth did not change
+	if hash == kymaSub.Status.Backend.WebhookAuthHash {
+		return nil
+	}
+
+	// pause subscription
+	state := types.State{Action: types.StateActionPause}
+	resp, err := em.client.UpdateState(eventMeshSub.Name, state)
+	if err != nil {
+		return fmt.Errorf("failed to pause EventMesh subscription: %w", err)
+	}
+	if resp.StatusCode == http.StatusNotFound {
+		em.namedLogger().Warnf("failed to pause EventMesh subscription: %s, not found", eventMeshSub.Name)
+		return nil
+	}
+	if resp.StatusCode != http.StatusAccepted {
```

Comment:
If subscription is already paused then will the request fail or will it still return `http.StatusAccepted`?